backend/Models/Groups/group-model.js

```diff
@@ -0,0 +1,434 @@
+ * @param {String} param0.username
+ * @return {Promise}
+ */
+
+function getAllGroupsOfUser({
+    username : username,
+}) {
+    return new Promise(function(resolve, reject) {
+        pool.query(`SELECT * FROM  UserGroups uG INNER JOIN groups g ON g.id = uG.group_id WHERE username = ? AND confidential = 0`,
+      [
+        username,
+      ],
+          function(error, results) {
+            if (error) {
+              reject(error);
+              return;
+            }
+            resolve(results);
+          }
+        );
+      });
+}
+/**
+ *
+ * @param {Object} param0
+ * @param {String} param0.username
+ * @return {Promise}
+ */
+
+function getAllGroups({
+    username : username,
+}) {
+    return new Promise((resolve,reject) => {
+        pool.query(`SELECT * FROM groups WHERE (SELECT COUNT(username) FROM users WHERE (username=? AND admin=1))`,
+        [
+            username
+        ],function(error, results) {
+            if (error) {
+              reject(error);
+              return;
+            }
+            resolve(results);
+          }
+        );
+      });
+}
+/**
+ *
+ * @param {Object} param0
+ * @param {String} param0.username
+ * @param {Number} param0.group_id
+ * @return {Promise}
+ */
+
+function getGroupById({
+    username,
+    group_id:group_id, 
+    }) {
+        return new Promise((resolve,reject) => {
+            pool.query(`SELECT * FROM groups WHERE id = ? AND (SELECT COUNT(username) FROM UserGroups WHERE (username=? AND group_id=?))`,
+            [
+                group_id,
+                username,
+                group_id,
+
+            ],function(error, results) {
+                if (error) {
+                  reject(error);
+                  return;
+                }
+                resolve(results);
+              }
+            );
+          });
+    }
+/**
+ *
+ * @param {Object} param0
+ * @param {String} param0.username
+ * @param {String} param0.description
+ * @param {Number} param0.group_id
+ * @return {Promise}
+ */
+function updateGroup({
+    username,
+    group_id:group_id,
+    description : description,
+}) {
+    
+        return new Promise((resolve,reject) => {
+            pool.query(`UPDATE groups SET description=? WHERE
+            (SELECT COUNT(username) FROM UserGroups WHERE (username=? AND admin=1 AND group_id = ?))`,
+            [
+                description,
+                username,
+                group_id,
+
+            ],function(error, results) {
+                if (error) {
+                  reject(error);
+                  return;
+                }
+                resolve(results);
+              }
+            );
+          });
+}
+/**
+ *
+ * @param {Object} param0
+ * @param {String} param0.username
+ * @param {Number} param0.group_id
+ * @return {Promise}
+ */
+function deleteGroup({
+    username,
+    group_id : group_id,
+}) {
+   return new Promise((resolve,reject) => {
+    pool.query(`DELETE FROM groups WHERE id = ? AND
```

Comment:
I would suggest, you create a deleted column in groups and set that to 1 when user deletes the group. We should avoid delete statements as far as possible.

Reply:
Fixed